tiledb/CMakeLists.txt

```diff
@@ -410,9 +410,12 @@ add_library(TILEDB_CORE_OBJECTS OBJECT
     ${TILEDB_EXTERNALS_SOURCES}
 )
 
-# Use BUILD_INTERFACE generator to work around
-#   https://gitlab.kitware.com/cmake/cmake/-/issues/20736
-target_link_libraries(TILEDB_CORE_OBJECTS PRIVATE $<BUILD_INTERFACE:common>)
+target_link_libraries(TILEDB_CORE_OBJECTS
+  PRIVATE
+    common
+  PUBLIC
+    baseline $<TARGET_OBJECTS:baseline>
```

Comment:
We'll be needing to add a number of object modules to both the dynamic and static libraries. In order to avoid synchronizing two lists, we make another object library that incorporates everything. We can use the `object_library` environment to define it in this file.

Reply:
I am actually planning to merge `tiledb_shared` and `tiledb_static` after we stop bundling the shared libraries on GitHub Releases, which should be unblocked soon (whether to build a static or dynamic library will be determined by the `BUILD_SHARED_LIBS` variable). After that, the object library that incorporates everything could be `TILEDB_CORE_OBJECTS`.

It wouldn't hurt to keep this duplication for a little while, right? I can add a comment.

Reply:
I tried moving the `baseline` dependency to `TILEDB_CORE_OBJECTS_ILIB` to remove duplication, but it fails:

```diff
diff --git a/tiledb/CMakeLists.txt b/tiledb/CMakeLists.txt
index ef97cb5c5..92b50f1ff 100644
--- a/tiledb/CMakeLists.txt
+++ b/tiledb/CMakeLists.txt
@@ -414,8 +414,6 @@ add_library(TILEDB_CORE_OBJECTS OBJECT
 target_link_libraries(TILEDB_CORE_OBJECTS
   PRIVATE
     common
-  PUBLIC
-    baseline $<TARGET_OBJECTS:baseline>
 )

 target_link_libraries(TILEDB_CORE_OBJECTS INTERFACE object_store_definitions)
@@ -470,7 +468,7 @@ if (WIN32 AND TILEDB_STATIC)

   target_link_libraries(TILEDB_CORE_OBJECTS_STATIC
     PUBLIC
-      baseline $<TARGET_OBJECTS:baseline>
+      baseline
   )

   target_compile_definitions(TILEDB_CORE_OBJECTS_STATIC
@@ -524,6 +522,11 @@ endif()
 # so we can use the targets created by the calls to find_package().
 add_library(TILEDB_CORE_OBJECTS_ILIB INTERFACE)

+target_link_libraries(TILEDB_CORE_OBJECTS_ILIB
+  INTERFACE
+    $<BUILD_INTERFACE:baseline> $<TARGET_OBJECTS:baseline>
+)
+
 # object store definitions
 target_link_libraries(TILEDB_CORE_OBJECTS_ILIB INTERFACE object_store_definitions)

@@ -1070,7 +1073,7 @@ endif()
 # Link the dependencies specified earlier
 target_link_libraries(tiledb_shared
   PRIVATE
-    $<TARGET_PROPERTY:TILEDB_CORE_OBJECTS_ILIB,INTERFACE_LINK_LIBRARIES>
+    TILEDB_CORE_OBJECTS_ILIB
     TILEDB_CORE_OBJECTS
 )

@@ -1121,7 +1124,7 @@ if (TILEDB_STATIC)

   target_link_libraries(tiledb_static
     PRIVATE
-      $<TARGET_PROPERTY:TILEDB_CORE_OBJECTS_ILIB,INTERFACE_LINK_LIBRARIES>
+      TILEDB_CORE_OBJECTS_ILIB
   )
 endif()
```